### DISCUSSION TITLE
Security vulnerabilities in log4j2, CVE-2021-44228

### DISCUSSION THREAD

**Discussion by AngryMills** [apache/skywalking#8280] [Q&A] [answered]
https://twitter.com/P0rZ9/status/1468949890571337731
Has any idea about this?
When will fix it?
Thanks for answering.

**Comment by wu-sheng** [comment, marked as answer]
## 8.9.1 is released

https://skywalking.apache.org/events/release-apache-skywalking-apm-8-9-1/

Upgrade log4j2 to 2.15.0 for CVE-2021-44228. This CVE only effects on JDK versions below 6u211, 7u201, 8u191 and 11.0.1 according to the post. Notice, using JVM option -Dlog4j2.formatMsgNoLookups=true also avoids CVE if your JRE opened JNDI in default.

**Comment by wu-sheng** [comment]
BTW, 2.15 is out, with JNDI disabled in default. Justb pumping up version should work.

**Reply by AngryMills** [reply]
thanks

**Reply by wu-sheng** [reply]
https://github.com/apache/skywalking/pull/8281 Patch has been submitted with description, this is just for `debug` logs in prod. I think it would be in the real world.

**Comment by wu-sheng** [comment]
CVE-2021-44228 has been officially reported and fixed by log4j2 2.15.0. I am running a 8.9.1 release to just include this dependency upgrade.
